codebase/codebase_test.go

```diff
@@ -46,29 +49,34 @@ func TestNewCodebase(t *testing.T) {
 	assert.Equal(t, "", cb.Branch)
 	assert.Equal(t, "", cb.FileName)
 	assert.Equal(t, 0, cb.LineNumber)
+	assert.Equal(t, "", cb.StackID)
 
 	// test for all values in codebase
 	branch := "task-101"
 	repo := "golang-project"
+	stackID := "golang-default"
 	file := "main.go"
 	line := 200
 	codebaseMap = map[string]interface{}{
 		codebase.RepositoryKey: repo,
 		codebase.BranchKey:     branch,
 		codebase.FileNameKey:   file,
 		codebase.LineNumberKey: line,
+		codebase.StackIdKey:    stackID,
 	}
 	cb, err = codebase.NewCodebaseContent(codebaseMap)
 	require.Nil(t, err)
 	assert.Equal(t, repo, cb.Repository)
 	assert.Equal(t, branch, cb.Branch)
 	assert.Equal(t, file, cb.FileName)
 	assert.Equal(t, line, cb.LineNumber)
+	assert.Equal(t, stackID, cb.StackID)
```

Comment:
StackID is unknown at WI level i think. It doesn't hurt to have it there, but technically for Recommender to have parsed this repository it would have been imported/created via Forge/Codebases and the StackID is already known. Recommender has no notion of stack id